VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/VitruvioComponent.cpp

```diff
@@ -648,40 +648,37 @@ void UVitruvioComponent::OnPropertyChanged(UObject* Object, FPropertyChangedEven
 	{
 		return;
 	}
+	bool bComponentPropertyChanged = false;
 
 	// Happens for example during import from copy paste
 	if (!PropertyChangedEvent.Property)
 	{
-		return;
-	}
-
-	bool bComponentPropertyChanged = false;
-
-	if (PropertyChangedEvent.Property->GetFName() == GET_MEMBER_NAME_CHECKED(UVitruvioComponent, Rpk))
-	{
-		Attributes.Empty();
-		bAttributesReady = false;
-		bComponentPropertyChanged = true;
-		bNotifyAttributeChange = true;
-	}
+		if (PropertyChangedEvent.Property->GetFName() == GET_MEMBER_NAME_CHECKED(UVitruvioComponent, Rpk))
+		{
+			Attributes.Empty();
+			bAttributesReady = false;
+			bComponentPropertyChanged = true;
+			bNotifyAttributeChange = true;
+		}
 
-	if (PropertyChangedEvent.Property->GetFName() == GET_MEMBER_NAME_CHECKED(UVitruvioComponent, RandomSeed))
-	{
-		bValidRandomSeed = true;
-		bComponentPropertyChanged = true;
-	}
+		if (PropertyChangedEvent.Property->GetFName() == GET_MEMBER_NAME_CHECKED(UVitruvioComponent, RandomSeed))
+		{
+			bValidRandomSeed = true;
+			bComponentPropertyChanged = true;
+		}
 
-	if (PropertyChangedEvent.Property->GetFName() == GET_MEMBER_NAME_CHECKED(UVitruvioComponent, GenerateCollision))
-	{
-		bComponentPropertyChanged = true;
-	}
+		if (PropertyChangedEvent.Property->GetFName() == GET_MEMBER_NAME_CHECKED(UVitruvioComponent, GenerateCollision))
+		{
+			bComponentPropertyChanged = true;
+		}
 	
-	if (PropertyChangedEvent.Property->GetFName() == GET_MEMBER_NAME_CHECKED(UVitruvioComponent, HideAfterGeneration))
-	{
-		InitialShape->SetHidden(HideAfterGeneration && HasGeneratedMesh);
+		if (PropertyChangedEvent.Property->GetFName() == GET_MEMBER_NAME_CHECKED(UVitruvioComponent, HideAfterGeneration))
+		{
+			InitialShape->SetHidden(HideAfterGeneration && HasGeneratedMesh);
+		}
 	}
 
-	const bool bRelevantProperty = InitialShape && InitialShape->IsRelevantProperty(Object, PropertyChangedEvent);
```

Comment:
With this change we now recreate the initial shape any time there is a property change on the actor. 

The idea was that we do not always recreate the initial shape/regenerate for all changes. Eg. when we move a spline point (I think there is actually a small bug with that in the `IsRelevantProperty` of the SplineInitialShape. The `PropertyChangedEvent.ChangeType == EPropertyChangeType::ValueSet` should always be checked and not only for the "SplineApproximationPoints").